### ISSUE TITLE
Flightmode not set correctly

### ISSUE THREAD

**Issue by Bre281** [renyuneyun/Easer#318]
I try to write a script to deactivate flightmode every 30 minutes for 3 minutes. But Easer doesn't seem to set flightmode correctly.

![screen](https://user-images.githubusercontent.com/60109817/72746990-64836300-3bab-11ea-994f-a34c8fd69aa7.png)

As you can see flightmode is activated, but there is still data connection.
I'm using LineageOS 16.0 with microG on a LG-D802 and Easer 0.7.9.1

**Comment by Bre281** [comment]
Forgot to mention: It doesn't happen all the time. Only sometimes :-(

**Comment by renyuneyun** [comment]
Hi there,
Thanks for reporting. But unfortunately I don't think there is a direct way to solve this, because basically Android doesn't allow an app to change flight mode. The current way it works is actually a workaround and does not guarantee to work on all devices.

However, the behaviour on your case is quite interesting/strange. I thought it should either work or not work.
Do you identify any other related information (e.g. does the "not changing flight mode" only happen on enabling but not on disabling)?

**Comment by Bre281** [comment]
It does happen on enabling and disabling. But when it happens seem to be random. Sometimes flight mode doesn't change for two or three "loops" of the script, then it works a while again.

Too bad that it can't be fixed easily. I helped myself now by switching mobile data on and off instead of flight mode. Should be a standard Android feature in my opinion, because not being online all the time saves lots of battery power.

BTW: What's the difference between exact and inexact timers? How more exact is the exact timer and is the battery drain much higher?

**Comment by renyuneyun** [comment]
Yes, that's very unconvinent but it's reasonable for the Android team to decide not allowing changing airplane mode (and mobile data actually) from a user app, in order to stop abuse.

Exact should be perfectly exact, according to Android documentation; inexact is adjusted (aligned) by Android system and even the developers (i.e. me) have no idea how inexact it will be (but hopefully within 10 minutes?).
The battery will drain faster to use exact, although the precise different differs from device to device, possibly (from my understanding) because of different installed apps to be woken up and the system (ROM) overhead.

**Comment by renyuneyun** [comment]
By the way, you can have a look at #261 if you want some more information about how airplane mode is currently being handled.
